docs/glossary.asciidoc

```diff
@@ -2,11 +2,15 @@
 == Glossary of {ecs} Terms
 
 [[glossary-ecs]] 
-ECS ::
-
+ECS::
 Elastic Common Schema. A common set of document fields, field names, and their respective entity
 relationships to be used in the storage of log messages and other data in
 Elasticsearch.
 
+ECS compatible::
+Describes an implementation with all existing fields mapped to ECS fields.
+
+ECS compliant:: Describes an implementation with all existing fields mapped to
+ECS fields, and with as many ECS core fields populated as is practical.
```

Comment:
I think we wanted to remove these two distinctions, right @MikePaquette?

These concepts haven't been fully fleshed out, I would remove for now, and only introduce when we're ready to have these kinds of discussions.

Reply:
@webmat @karenzone 
I agree to remove these definitions for now, but I want to make sure we include guidance like: "In order to increase the level of interoperability of your data with future analysis content, your implementation should attempt to populate as many of the Core fields as practical, even if these fields or values are metadata that are not present in your original event, or are already populated in another ECS field"

I think this guidance is fundamental to the success of a common schema, and one I've not been very successful at articulating.

Simple example: when converting a syslog firewall log event to ECS, there maybe no field in the event with the value "syslog" , but a good ECS implementation  will still populate `agent.type: syslog` so that future ECS-aware analysis content that filters events by `agent.type` will work properly.

Reply:
I've taken an additional note to reinforce the docs this way, in the docs plan